rezz.lic

```diff
@@ -1,119 +1,126 @@
+			  echo "Using default values for REZZ, make a 'rezz'=>Resurrection waggle set to avoid this."
+			  get_data('spells').spell_data['Resurrection']
+			end
+
+    @rejuv = if @settings.waggle_sets['rezz']['Rejuvenation']
+	           @settings.waggle_sets['rezz']['Rejuvenation']
+			 else
+			   echo "Using default values for REJUV, make a 'rezz'=>Rejuvenation waggle set to avoid this."
+	           get_data('spells').spell_data['Rejuvenation']
+			 end
+
+    Flags.add('no-favor', /has no favor with h.. god!/)
+    Flags.add('rejuv-silver', 'A thin silver nimbus surrounds', 'As you concentrate, a thin silver nimbus flickers into view around')
+    Flags.add('alive', 'Nothing happens')
+    Flags.add('necro', 'has no protection from the gods. judgment')
+
+    @target.each do |name|
+      @rejuv['cast'] = "cast #{name}"
+	  rejuv_process(name)
+      Flags['alive'] ? begin echo "*** #{name} is already alive, dumbass! ***"; next end : nil
+      Flags['no-favor'] ? begin echo "*** #{name} has no favors! ***"; next end : nil
+      rezz_process(name)
+	end
   end
 
-  def rezz_prep(person)
-    # cast buffs if set
-    check_buffs
-
-    # perceive body & rejuv
-    rejuv(person)
-
-    # cast rezz
-    mana = cast_rezz
-
-    # infuse untill see spirit
-    find_soul(person, mana)
-
-    # snap cast soul bonding
-    cast_soulbonding(person)
-
-    # check for silver again. recast rejuv (or infuse again) if not
-    rejuv(person)
-
-    # gesture
-    bput("gest #{person}", 'Roundtime')
+  def find_dead
+    /You take a moment to look for everybody in the area and see (.*)/ =~ bput('look people', "You look around and notice that you're the only one in the area", /You take a moment to look for everybody in the area and see (.*)/)
+    @target = Regexp.last_match(1).sub(/ and /, ', ').scan(/the body of ([a-z[:blank:]]+) who is [a-z[:blank:]]+?,?./i).flatten.map { |obj| obj.split.last }
+	@target == [] ? @target = nil : nil
   end
 
-  def find_soul(person, infuse_amt)
-    loop do
-      pause 5 while mana <= 40
-      harness_mana([infuse_amt]) unless skip_harness?
-      case bput("infuse rezz #{infuse_amt}", /You clearly sense the spirit of #{person} in the Void/i, 'You sense a spirit nearby, but you are unable to make it out clearly.', 'Your sense of sight has not fully transitioned to the spirit world', "You don't have enough harnessed mana to infuse that much.")
-      when 'You sense a spirit nearby, but you are unable to make it out clearly.'
-        next
-      when /You clearly sense the spirit of #{person} in the Void/i
-        break
-      when 'Your sense of sight has not fully transitioned to the spirit world'
-        pause 2
-        next
-      when "You don't have enough harnessed mana to infuse that much."
-        next
-      else
-        next
-      end
+  def rejuv(player, quick = false)
+    Flags.reset('rejuv-silver')
+    quick ? begin @rejuv['cambrinth'] = []; @rejuv['prep_time'] = 5; @rejuv['mana'] = 10 end : nil
```

Comment:
I'm going to be honest. These if checks for begins are slick, but they are also tough to read, maybe an occasional use would be OK, but this many is becoming overused.

Reply:
Could be a personal preference thing too, not entirely sure. :D

Reply:
I was just trying to keep the code condensed.  I have no issues writing them out in normal blocks.

I like those ternary if statements, but they don't work if you need to do more than 1 command in either section.  I found some example with the begin/end and used it.  But the style guide says that it's a bad idea for code readability.

Reply:
Yeah, they're slick but could become cumbersome to read. I think this is all personal preference though. Keep doing it, I would only care ultimately if the code owners cared.

Reply:
Yeah.  I like those ternary ifs, but after reading a little of the ruby style guide it says that doing what I did is bad.  I'll change it.